File: test/bench/vs/bench-sdb.py

from datetime import datetime
from sdb import *

db = Sdb (".","lala.db",None)

time_a = datetime.now()
for a in range(0,100000):
	db.set("foo"+str(a), "bar", 0)
time_b = datetime.now()
print (time_b-time_a)
db.sync ()
time_c = datetime.now()
print (time_c-time_b)

File: test/bench/vs/bench-sdb.c
#include <sdb/sdb.h>

int main() {
	int i;
	char foo[32];
	Sdb *s = sdb_new (".","___test.sdb",0);
	for (i=0;i<100000; i++) {
		snprintf (foo, sizeof (foo), "foo%d", i);
		sdb_set (s, foo, "bar", 0);
	}
	sdb_sync (s);
	sdb_free (s);
	return 0;
}
